tests/integration/targets/aci_bd_to_l3out/tasks/main.yml

```diff
@@ -0,0 +1,202 @@
+# Test code for the ACI modules
+# Copyright: (c) 2023, Gaspard Micol (@gmicol) <gmicol@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: debug
+
+# CLEAN ENVIRONMENT
+- name: Remove the ansible_tenant
+  cisco.aci.aci_tenant: &aci_tenant_absent
+    <<: *aci_info
+    state: absent
+    tenant: ansible_test
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Create tenant for tests to kick off
+    cisco.aci.aci_tenant: &aci_tenant_present
+      <<: *aci_tenant_absent
+      state: present
+
+  - name: Create vrf for tests to kick off
+    cisco.aci.aci_vrf: &aci_vrf_present
+      <<: *aci_tenant_present
+      vrf: anstest
+
+  - name: Create first L3out for tests to kick off
+    cisco.aci.aci_l3out: &aci_l3_out_present
+      <<: *aci_vrf_present
+      l3out: ansible_l3out
+      domain: ansible_dom
+      state: present
+
+  - name: Create second L3out for tests to kick off
+    cisco.aci.aci_l3out: &aci_l3_out_present_2
+      <<: *aci_vrf_present
+      l3out: ansible_l3out_2
+      domain: ansible_dom
+      state: present
+
+  - name: Create first bd for tests to kick off
+    cisco.aci.aci_bd: &aci_bd_present
+      <<: *aci_tenant_present
+      bd: anstest
+
+  - name: Create second bd for tests to kick off
+    cisco.aci.aci_bd: &aci_bd_present_2
+      <<: *aci_tenant_present
+      bd: anstest_2
+
+  - name: Ensure first binding bd_to_l3out does not exist for tests to kick off
+    cisco.aci.aci_bd_to_l3out: &aci_bd_to_l3out_absent
```

Comment:
change

    name: ensure tenant exists for tests to kick off. -> - name: Create tenant

Here you are adding the tenant (not testing if it exists).

Everywhere else this applies.